src/backend/distributed/commands/multi_copy.c

```diff
@@ -1751,6 +1752,12 @@ CitusCopyDestReceiverStartup(DestReceiver *dest, int operation,
 	LockShardListMetadata(shardIntervalList, ShareLock);
 	LockShardListResources(shardIntervalList, ShareLock);
 
+	/*
+	 * We might have some concurrent metadata changes and we need to accept the cache
+	 * invalidation messages to learn about the changes.
+	 */
+	AcceptInvalidationMessages();
```

Comment:
> For reloading the cache entry, I was also thinking to add to make the code more readable. 

Yep, seems like a good idea. Note that the `copyDest->tableMetatada = cacheEntry` is completely safe since the `cacheEntry` pointer always stays the same.

What's slightly more suspicious is that we do `cacheEntry->replicationModel == REPLICATION_MODEL_2PC` right after a possible invalidation. Theoretically, we might see a stale replication model here :).